test/TokenVendor.t.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {Vm} from "forge-std/Vm.sol";
+import {Test} from "forge-std/Test.sol";
+
+import {Token} from "../src/Token.sol";
+import {TokenVendor} from "../src/TokenVendor.sol";
+
+contract TokenVendorTest is Test {
+    Token token;
+    TokenVendor tokenVendor;
+
+    address admin;
+    address minter;
+
+    function setUp() public {
+        vm.createSelectFork("https://gateway.tenderly.co/public/sepolia");
+
+        admin = makeAddr("admin");
+        minter = makeAddr("minter");
+
+        token = new Token(admin, minter);
+        tokenVendor = new TokenVendor(address(token));
+
+        // Grant tokenVendor minter role.
```

Comment:
Note that deployment arguments are not perfect. Token depends on TokenVendor for the minter address, TokenVendor on Token for the token address. However, it works :)